pilot/pkg/networking/core/listener_waypoint.go

```diff
@@ -473,6 +473,12 @@ func (lb *ListenerBuilder) buildWaypointInternal(wls []model.WorkloadInfo, svcs
 		}
 	}
 	tlsInspector := func() *listener.ListenerFilter {
+		// If we have enrolled workloads, we should sniff TLS
+		// This is important if users want to assert authz against SNI.
```

Comment:
Does authz policy even allow this today?

Reply:
when do we support sni in authorizationPolicy